plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/parser/rules/ScriptParameterRule.java

```diff
@@ -117,7 +118,33 @@ && isIdentifierQuote(buffer.charAt(buffer.length() - 1), false, true)
         return TPTokenAbstract.UNDEFINED;
     }
     
-    private boolean isIdentifierQuote(char c, boolean testStart, boolean testEnd) {
+    public static int tryConsumeParameterName(SQLDialect sqlDialect, CharSequence buffer, int position) {
+        // keep in sync with the above
+        if (position >= buffer.length()) {
+            return -1;
+        }
+        String[][] quoteStrings = sqlDialect.getIdentifierQuoteStrings();
+        char c = buffer.charAt(position);
+        position++;
+        if (isIdentifierQuote(c, true, false, quoteStrings)) {
+            while (position < buffer.length()) {
+                c = buffer.charAt(position);
+                position++;
+                if (isIdentifierQuote(c, false, true, quoteStrings)) {
+                    return position;
+                }
+            }
+            return -1;
+        } else {
+            while (position < buffer.length() && Character.isJavaIdentifierPart(c)) {
```

Comment:
Can we rely on `SQLDialect#validIdentifierStart` and `SQLDialect#validIdentifierPart` instead?

Reply:
We parse unquoted variable name here, so it will not be correct to do so.